### ISSUE TITLE
variable assignment after systemCommand is ignored

### ISSUE THREAD

**Issue by hemmecke** [fricas/fricas#146]
Suppose there is a file `foo.input` with the following content.
```
foo:="bar";
```
Putting the variable assignment of` var`directly after the `systemCommand`, then it seems to be ignored.
```
(1) -> systemCommand("read foo.input")$MoreSystemCommands; var := foo; 
foo:="bar";
 

                                                                 Type: String
(2) -> var

   (2)  var
                                                          Type: Variable(var)
(3) -> foo

   (3)  "bar"
                                                                 Type: String
```

**Comment by hebisch** [comment]
On Thu, Dec 07, 2023 at 12:58:14AM -0800, Ralf Hemmecke wrote:
> Suppose there is a file `foo.input` with the following content.
> ```
> foo:="bar";
> ```
> Putting the variable assignment of` var`directly after the `systemCommand`, then it seems to be ignored.

'read' command aborts current line and passes control to next
line.  For example:

systemCommand("read foo.input")$MoreSystemCommands; print("read OK");

prints nothing.  ATM this seem to be necessary to properly
reading.

-- 
                              Waldek Hebisch

**Comment by hemmecke** [comment]
> systemCommand("read foo.input")$MoreSystemCommands; print("read OK");

Even though it is explainable, it is clearly a bug.

I would have thought that the interpreter could split such a line above 
at the ";" and take the rest as the "next line" internally. Or does this 
confuse line counting or something else?

**Comment by hemmecke** [comment]
Now, this bug seems to bite me. Well, a solution might be to generate a big .input file with all the different read command in them, but I would actually like to do the reading of several files in a loop like this.
```
for pow in 1..3 repeat
  dbgPrint("pow", pow)
  sys(concat["read ", fn, string(pow),".input )quiet"])
  dbgPrint("POW", pow)
```
Above I do not even consider the the side effects that I actually want from the above, but it only prints
```
   "powA1"
```
Maybe, someone has a quick suggestion how I can run a loop with a "read" inside.

**Comment by hemmecke** [comment]
Oh, to which mail (url?) do you refer? I must have overlooked it and don't find it at fricas-devel.
Anyway, I've removed the two lines from `i-syscmd.boot` so that I now have
```
do_read(ll, quiet, pile_mode) ==
    $nopiles : local := pile_mode
    $edit_file := ll
    read_or_compile(quiet, false)
```
Then ran the following:
```
)read i-syscmd.boot
for pow in 1..3 repeat (_
  dbgPrint("pow", pow);_
  cmd := concat["read ", fn, string(pow),".input )quiet"];_
  dbgPrint("cmd", cmd);_
  sys(concat["read ", fn, string(pow),".input )quiet"]);_
  vhl.pow := hl;_
  dbgPrint("POW", pow))
```
Well, the "read" and the loop work nicely, BUT it never returns to the commandline prompt.
I have to press Ctrl-C to get my prompt back after the "for".
However, the vector `vhl` seems to be correctly filled by the loop.
The last thing that is printed is "POW"
In other words the patch is not complete.

If I change the last line to
```
  dbgPrint("POW", pow)) ; error "blah"
```
of course the prompt comes back. But that is a hack.

**Comment by hebisch** [comment]
On Mon, Dec 18, 2023 at 01:03:57PM -0800, Ralf Hemmecke wrote:
> Oh, to which mail (url?) do you refer? I must have overlooked it and don't find it at fricas-devel.

My first message in thread "Recursive call to interpreter" (Message-ID:
 ***@***.***>) contained a patch.  Later in the
same thread Qian posted his series of patches.

> Anyway, I've removed the two lines from `i-syscmd.boot` so that I now have
> ```
> do_read(ll, quiet, pile_mode) ==
>     $nopiles : local := pile_mode
>     $edit_file := ll
>     read_or_compile(quiet, false)
> ```

<snip>

> In other words the patch is not complete.

Yes, see above.

-- 
                              Waldek Hebisch

**Comment by oldk1331** [comment]
Should be fixed by 6d19ee82c9d702d96c586ac4b4e923b5e63ec1c9 .
